server/server.ts

```diff
@@ -243,7 +271,11 @@ export default async function (
 				performAuthentication.call(socket, {});
 			} else {
 				socket.on("auth:perform", performAuthentication);
-				socket.emit("auth:start", serverHash);
+				socket.emit("auth:start", {
+					serverHash,
+					openidEnabled: Config.values.openid.enable && !Config.values.public,
+					openidInit: issuerURL,
+				});
```

Comment:
This doesn't seem right.
that event should know nothing about specific plugins.
I don't think that there should be any kind of openid data in there

This is coupling the server to the openid plugin, which defeats the purpose of plugins in the first place.

The point of using plugins is that the server (ideally, pretty sure we violate it already) doesn't know anything about how auth works and what the special methods need.

Reply:
Auth has to be triggered by the user, not the server. As such, the client has to know if OpenID is enabled and where to redirect the user.

Reply:
The auth is controlled by the server, not the user.
What you are doing there is mixing two unrelated event types, namely openid + the hash auth into one, flagged by a bool.

That's not really how you should do it.

If a auth backend needs special logic, then pass control to the auth backend and let it return the data. Then strongly type it.

The client should get an event and already know what it is, not some grab bag that contains maybe a hash, maybe openid connect, maybe $next_huge_auth_thing etc.

That just proliferates into a mess